### ISSUE TITLE
CSS should be more consistent with "classical" JLab for now

### ISSUE THREAD

**Issue by fperez** [executablebooks/jupyterlab-myst#61]
### Describe the bug

**context**

Since this is a new renderer that we'll want to get buy-in from the community for, my suggestion would be to first keep the CSS/layout changes to an absolute minimum, focusing only on implementing support for new features of MyST that extend beyond what the default renderer offers.

Only once we reach parity on all core features + support for the unique MyST ones, should we explore in earnest modifications to the rendering behavior.

**bug**

These aren't strictly bugs, but I will list them here for starters. I'm happy to track this elsewhere/differently if it better fits the team! In all these screenshots, the left shows the "classic" JupyterLab renderer, with the mystjs 0.1.3 renderer on the right. Both are using JLab 3.6.1.

* Code cells use a different/smaller font. 

![image](https://user-images.githubusercontent.com/57394/217388753-05e3bf86-5fa6-4aa6-bbab-09cdef9569cb.png)

* Missing outline around active cell.

![image](https://user-images.githubusercontent.com/57394/217388989-dab481df-5e8e-4bea-9bc0-ab2c42239e27.png)

* Too much vertical space used. I think it's mostly due to extra vertical whitespace around headings, though there's also other font choices that have an impact, such as larger fonts for headings. I actually happen to like those, but I imagine they could be controversial, so again, perhaps better staged for later discussion separate from the core features of the renderer.

![image](https://user-images.githubusercontent.com/57394/217390270-40c833fe-b683-4835-b13a-bcec9d34c11b.png)

* Style changes in rendering of inline unformatted text:

![image](https://user-images.githubusercontent.com/57394/217390343-0e9fba6b-b44f-44d0-b5a6-33e85a310774.png)

* Same, but when that text is in a heading:

![image](https://user-images.githubusercontent.com/57394/217390610-efe6e590-84a5-4986-9281-1547477fbe77.png)

* Changes to rendering of code blocks. I actually like these a lot, so I'd be tempted to consider them "pure improvements" (esp. the addition of the little "copy" button, which I _love_). But perhaps in the interest of making this as easy to adopt/discuss with the community for adoption first as a "better renderer" we should keep the differences to an absolute minimum, and discuss rendering updates separately:

![image](https://user-images.githubusercontent.com/57394/217390927-f7d132bb-45e3-4425-9ce9-db2b2d522b5b.png)

### Reproduce the bug

All these can be seen with 0.1.3.

**Comment by tavin** [comment]
I've encountered some perhaps related buggy rendering in connection with headings:

![image](https://user-images.githubusercontent.com/7685034/222001682-0382e65a-9b1e-4421-9700-b63ef6d8a4bd.png)

The first heading level element is changed to h1, and moved above any preceding content.

After that, h2 level elements get so much extra vertical margin/padding they end up taking more space than if they were h1 level.

The screenshot is from jupyterlab 3.6.1 and jupyterlab_myst 1.1.1.

**Comment by rowanc1** [comment]
Thanks @tavin  -- the code is currently pulling out the first cell title to create a frontmatter block. I think this is probably unexpected behaviour as you were surprised, I will turn that into a bug in mystjs and get it fixed.

Thanks again for these comment on the styles, I want to take a more in-depth pass here soon to get things inline with expectations!

**Comment by tavin** [comment]
Hi @rowanc1 nice the heading/frontmatter issue will be fixed. In the meantime I've just learned that adding yaml frontmatter is a pretty decent workaround, thanks to your answer.

Now the top margin on the h2 level elements seems to be unrelated and indeed a CSS issue, in fact an explicit `margin-top:2em` coming from a myst style.

**Comment by tavin** [comment]
@rowanc1 @fperez taking a stab at normalizing things: #114

**Comment by rowanc1** [comment]
Quite a few fixes from @tavin coming in for this in various places! Thanks for all your help @tavin.

I have picked up your changes in #114 and made an additional push for more Jupyter-classic styles in #116. With that coming in, I think this addresses all points (and a few others) raised in this thread. The only exception to that right now is the code-renderer, which still has the shadow/border/copy-button - that requires a deeper change.

**Comment by fperez** [comment]
BTW - if those are actually more work to remove, I'd prioritize other work over "fixing"  the code renderer :) Other than being different, I actually find it to be a net positive overall (both æsthetically and functionally), so perhaps we fry other fish first!

**Comment by rowanc1** [comment]
Many of these changes have been released in [v1.1.3](https://github.com/executablebooks/jupyterlab-myst/releases/tag/v1.1.3). Thanks @tavin @agoose77 and @fperez for the help in documenting the bugs, and jumping in to get this over the line! 🚀

@fperez looking forward to a new round of screenshots/improvements to make (either here or in a new issue!). I think one of the major next things is to get inline HTML fully working, and that will expose more things as well.
